### PR TITLE
feat(search): add COLLECT reducer for FT.AGGREGATE

### PR DESCRIPTION
**Pull request by ndyakov**
Add FTAggregateCollect options, NewCollectReducer constructor, and AggregateBuilder.Collect that render the REDUCE COLLECT clause (FIELDS/ DISTINCT/SORTBY/LIMIT) with automatic @-prefixing and arg-count. Add AggregateRow.Collect to decode the sparse RESP2/RESP3 entry array into a uniform CollectColumn. Integration-tested against Redis 8.8. Additive: no existing type or serializer changed.

Requires Redis 8.8+ with search-enable-unstable-features.

---

> [!NOTE]
> **Low Risk**
> Additive search API with no changes to existing serializers; requires Redis 8.8+ unstable features on the server.
> 
> **Overview**
> Adds client support for the **FT.AGGREGATE `COLLECT` GROUPBY reducer** (Redis 8.8+, behind `search-enable-unstable-features`), so aggregations can return each group plus projected/sorted/limited member rows in one query.
> 
> **Request building:** New `FTAggregateCollect` options, `NewCollectReducer`, and `AggregateBuilder.Collect` assemble `REDUCE COLLECT` with `FIELDS` / `DISTINCT` / `SORTBY` / `LIMIT`, automatic `@` normalization, and correct `narg`. `SearchCollect` is registered on the existing aggregator enum.
> 
> **Response decoding:** `AggregateRow.Collect(alias)` normalizes COLLECT columns from RESP2 (flat key/value arrays) and RESP3 (maps) into `[]map[string]interface{}`, including sparse entries.
> 
> Ships with unit tests (wire oracle + decode edge cases), a live-server integration test (RESP2/RESP3, version gating), and `example/search-aggregate-collect`. Existing aggregate serialization paths are unchanged aside from the new reducer name.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit a796f94323c55ed7d7bfea2567a32525beb40c24. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: search_collect.go [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+	Offset int
+	Count  int
+}
+
+// ensureAtPrefix normalizes a field or sort name to a single leading "@".
+func ensureAtPrefix(name string) string {
+	if strings.HasPrefix(name, "@") {
+		return name
+	}
+	return "@" + name
```
[the remark this answers is not in the document]
Comment (ndyakov) [reply search_collect.go]: this is correct

**Review by ofekshenawa — DISMISSED** [review]

**Review by ofekshenawa — DISMISSED** [review]

**Review by ofekshenawa — DISMISSED** [review]

**Review by ofekshenawa — APPROVED** [review]
